test/e2e/soak_k8petstore.go

```diff
@@ -0,0 +1,181 @@
+			if status, ok := exiterr.Sys().(syscall.WaitStatus); ok {
+				log.Printf("Exit Status: %d", status.ExitStatus())
+			}
+		}
+	}
+	Expect(err).NotTo(HaveOccurred())
+	Logf("... Done starting k8petstore ")
+
+	//Lets poll until we reach min_expected transactions
+	totalTransactions := 0
+	startupTimeoutSeconds := 500
+	Logf("Start polling, timeout is %v seconds", maxSeconds)
+
+	//How long until the FIRST transaction...
+	startupTimeout := time.After(time.Duration(startupTimeoutSeconds) * time.Second)
+
+	//How long we will wait until we reach the nth (ie. 500th) transaction
+	transactionsCompleteTimeout := time.After(time.Duration(maxSeconds) * time.Second)
+	tick := time.Tick(2 * time.Second)
+	var ready = false
+
+	Logf("Now waiting %v seconds to see progress (transactions  > 3)", startupTimeoutSeconds)
+T:
+	for {
+		select {
+		case <-transactionsCompleteTimeout:
+			//i.e. if we dont have n*1000 transactions, fail the test.
+			Logf("Timeout %v reached. Breaking!", tick)
+			break T
+		case <-startupTimeout:
+			err, totalTransactions = readTransactions(c, ns)
+			Logf("Timeout %v reached.  Checking if transactions have occured.", startupTimeout)
+			//If we don't have 3 transactions, fail the test.
+			if err != nil {
+				Logf("Failed : Error %v", err)
+				break T
+			}
+			if totalTransactions < 3 {
+				break T
+			}
+			ready = true
+		case <-tick:
+			//Pass if we've collected enough transactions
+			err, totalTransactions = readTransactions(c, ns)
+			if ready {
+				Expect(err).NotTo(HaveOccurred())
+			}
+			if totalTransactions > minExpected {
+				break T
+			}
+			Logf("%v == %v total petstore transactions stored into redis. ==", time.Now(), totalTransactions)
+		}
+	}
+
+	//Finally! We should have exceeded the min_expected num of transactions.
+	//If this fails, but there are transactions being created, we may need to recalibrate
+	//the min_expected value - or else - your cluster is broken/slow !
+	Ω(totalTransactions).Should(BeNumerically(">", minExpected))
+}
+
+var _ = Describe("K8 Pet Store", func() {
+
+	//Used for calibrating expected transaction #s in the soak portion.
+	var minionCount int
+	f := NewFramework("k8petstore")
+
+	//On a single node cluster, we expect about 10 transactions every second on average.
+	//admittedly, this is a very rough estimate given that the ETL is done in batches...
+	It("Should  scale to 50 transactions in one minute", func() {
+		minions, err := f.Client.Nodes().List(labels.Everything(), fields.Everything())
+		Expect(err).NotTo(HaveOccurred())
+		minionCount = len(minions.Items)
+
+		//max number of trial before k8petstore.sh dies.
+		var loadGenerators int = minionCount
+		var restServers int = minionCount
+		fmt.Printf("load generators / rest generators [ %v  /  %v ] ", loadGenerators, restServers)
+		//At least 500 transactions should be acquired within a short time period.
+		runK8petstore(restServers, loadGenerators, f.Client, f.Namespace.Name, 50, 60)
```

Comment:
Please make the 50 and 60 named configurable constants, as per previous review.  You'll need to fix the above "It" statement too, which contains duplicates of those numbers.